src/httpServer/server.ts

```diff
@@ -169,6 +171,24 @@ async checkPortAvailability(port: number, host: string): Promise<void> {
     dbosExec.logger.debug(`DBOS Server Registered Perf GET ${HealthUrl}`);
   }
 
+  /**
+   * Register Deactiviate endpoint.
+   * Deactivate consumers so that they don'nt start new workflows.
+   */
+  static registerDeactivateEndpoint(dbosExec: DBOSExecutor, router: Router) {
+    console.log("Registering DeactivateUrl", DeactivateUrl);
+    const deactivateHandler = async (koaCtxt: Koa.Context, koaNext: Koa.Next) => {
+      dbosExec.logger.info("Deactivating consumers");
+      await dbosExec.deactivateConsumers();
+      dbosExec.logger.info("Deactivating Scheduler");
+      await dbosExec.scheduler?.destroyScheduler();
+      koaCtxt.body = "Deactivated";
+      await koaNext();
+    };
+    router.get(DeactivateUrl, deactivateHandler);
+    dbosExec.logger.info(`DBOS Server deactivate GET ${DeactivateUrl}`);
+  }
```

Comment:
This deactivate all scheduled workflows, unregister all event consumers. So the usage flow asks you to deactivate every existing workflows, then re-deploy or re-activate them.

We should chat about what the ideal UX would be.

I think it is going to be fairly hard to use in practice and we should have the API accept workflow IDs / event names, for selective deactivation.

Reply:
This will be used transparently by DBOS Cloud to prevent old versions from starting new workflows.